parse/src/test/java/com/parse/EncryptedFileObjectStoreTest.java

```diff
@@ -0,0 +1,125 @@
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+package com.parse;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.isNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.skyscreamer.jsonassert.JSONAssert.assertEquals;
+
+import android.content.Context;
+
+import androidx.security.crypto.EncryptedFile;
+import androidx.security.crypto.MasterKey;
+import androidx.test.platform.app.InstrumentationRegistry;
+
+import org.json.JSONObject;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.skyscreamer.jsonassert.JSONCompareMode;
+
+import java.io.File;
+
+import kotlin.jvm.JvmStatic;
+
+@RunWith(RobolectricTestRunner.class)
+public class EncryptedFileObjectStoreTest {
+
+    @Rule
+    public final TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    @JvmStatic
+    @Before
+    public void setUp() {
+        RobolectricKeyStore.INSTANCE.getSetup();
+        ParseObject.registerSubclass(ParseUser.class);
+        Parse.initialize(new Parse.Configuration.Builder(InstrumentationRegistry.getInstrumentation().getTargetContext()).server("http://parse.com").build());
+    }
+
+    @After
+    public void tearDown() {
+        ParseObject.unregisterSubclass(ParseUser.class);
+    }
+
+    @Test
+    public void testSetAsync() throws Exception {
+        File file = new File(temporaryFolder.getRoot(), "test");
+
+        ParseUser.State state = mock(ParseUser.State.class);
+        JSONObject json = new JSONObject();
+        json.put("foo", "bar");
+        ParseUserCurrentCoder coder = mock(ParseUserCurrentCoder.class);
+        when(coder.encode(eq(state), (ParseOperationSet) isNull(), any(PointerEncoder.class)))
+            .thenReturn(json);
+        EncryptedFileObjectStore<ParseUser> store = new EncryptedFileObjectStore<>(ParseUser.class, file, coder);
+
+        ParseUser user = mock(ParseUser.class);
+        when(user.getState()).thenReturn(state);
+        ParseTaskUtils.wait(store.setAsync(user));
+
+        Context context = InstrumentationRegistry.getInstrumentation().getContext();
+        EncryptedFile encryptedFile = new EncryptedFile.Builder(context, file, new MasterKey.Builder(context).setKeyScheme(MasterKey.KeyScheme.AES256_GCM).build(), EncryptedFile.FileEncryptionScheme.AES256_GCM_HKDF_4KB).build();
+        JSONObject jsonAgain = ParseFileUtils.readFileToJSONObject(encryptedFile);
+        assertEquals(json, jsonAgain, JSONCompareMode.STRICT);
+    }
+
+    @Test
+    public void testGetAsync() throws Exception {
```

Comment:
Can we have a test covering the `migrate`? Pardon me if is covered already. If it is can you point it to me?